src/main/java/org/spongepowered/api/service/command/GameArguments.java

```diff
@@ -0,0 +1,282 @@
+     * @param game The game to find worlds from
+     * @return the argument
+     */
+    public static CommandElement world(Text key, Game game) {
+        return new WorldPropertiesCommandElement(key, game);
+    }
+
+    private static class WorldPropertiesCommandElement extends CommandElement {
+        private final Game game;
+
+        protected WorldPropertiesCommandElement(Text key, Game game) {
+            super(key);
+            this.game = game;
+        }
+
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            final String worldName = args.next();
+            Optional<WorldProperties> ret = this.game.getServer().getWorldProperties(worldName);
+            if (!ret.isPresent()) {
+                throw args.createError(t("Unable to find world for name %s", worldName));
+            } else if (!ret.get().isEnabled()) {
+                throw args.createError(t("World %s is not enabled!", worldName));
+            }
+            return ret.get();
+        }
+
+        @Override
+        public List<String> complete(CommandSource src, CommandArgs args, CommandContext context) {
+            final Optional<String> worldNameComponent = args.nextIfPresent();
+            Iterable<String> worldsList = Iterables.transform(this.game.getServer().getAllWorldProperties(), new Function<WorldProperties, String>() {
+                @Nullable
+                @Override
+                public String apply(@Nullable WorldProperties input) {
+                    return input == null || !input.isEnabled() ? null : input.getWorldName();
+                }
+            });
+            worldsList = Iterables.filter(worldsList, Predicates.notNull());
+            if (worldNameComponent.isPresent()) {
+                worldsList = Iterables.filter(worldsList, new StartsWithPredicate(worldNameComponent.get()));
+            }
+            return ImmutableList.copyOf(worldsList);
+        }
+    }
+
+    /**
+     * Expect an argument to represent a dimension.
+     * Gives values of tye {@link DimensionType}
+     *
+     * @param key The key to store under
+     * @param game The game to find dimensions from
+     * @return the argument
+     */
+    public static CommandElement dimension(Text key, Game game) {
+        return catalogedElement(key, game, DimensionType.class);
+    }
+
+    /**
+     * Expect an argument to represent a {@link Vector3d}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement vector3d(Text key) {
+        return new Vector3dCommandElement(key);
+    }
+
+    private static class Vector3dCommandElement extends CommandElement {
+
+        protected Vector3dCommandElement(Text key) {
+            super(key);
+        }
+
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            return null;
```

Comment:
Do you plan to implement this before the merge? If not, add a TODO comment for the missing stuff